packages/web-app-markdown-editor/src/App.vue

```diff
@@ -98,14 +143,44 @@ export default {
       return loadFileTask.isRunning || saveFileTask.isRunning
     })
 
+    const config = computed(() => {
+      const { showPreview = true } = getCurrentInstance().proxy.applicationConfig
+      return { showPreview }
+    })
+
     onMounted(() => {
-      loadFileTask.perform(getCurrentInstance().proxy)
+      const that = getCurrentInstance().proxy
+      loadFileTask.perform(that)
+
+      // Enable ctrl/cmd + s
+      document.addEventListener('keydown', handleSKey, false)
+      // Ensure reload is not possible if there are changes
+      window.addEventListener('beforeunload', handleUnload)
+    })
+
+    onBeforeUnmount(() => {
+      window.removeEventListener('beforeunload', handleUnload)
+      document.removeEventListener('keydown', handleSKey, false)
     })
 
     const save = function () {
       saveFileTask.perform(this)
```

Comment:
This seems weird, what is `this` in this context? I doubt it provides `putFileContents`, better pass `defaults` here as well